x-pack/libbeat/management/tests/fbtest/filebeat_v2_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package fbtest
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	fbroot "github.com/elastic/beats/v7/x-pack/filebeat/cmd"
+	// initialize the plugin system before libbeat does, so we can overwrite it properly
+	_ "github.com/elastic/beats/v7/x-pack/libbeat/management"
+	"github.com/elastic/beats/v7/x-pack/libbeat/management/tests"
+	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
+)
+
+var expectedFBStreams = &proto.UnitExpectedConfig{
+	DataStream: &proto.DataStream{
+		Namespace: "default",
+	},
+	Type:     "logfile",
+	Id:       "logfile-system-default-system",
+	Name:     "system-1",
+	Revision: 1,
+	Meta: &proto.Meta{
+		Package: &proto.Package{
+			Name:    "system",
+			Version: "1.17.0",
+		},
+	},
+}
+
+func TestFilebeat(t *testing.T) {
+	filebeatCmd := fbroot.Filebeat()
+	tests.InitBeatsForTest(t, filebeatCmd)
+	var fbStreams = []*proto.Stream{
+		{
+			Id: "logfile-system.syslog-default-system",
+			DataStream: &proto.DataStream{
+				Dataset: "system.syslog",
+				Type:    "logs",
+			},
+			Source: tests.RequireNewStruct(map[string]interface{}{
+				"paths":         []interface{}{"./testdata/messages"},
+				"exclude_files": []interface{}{".gz$"},
+				"multiline": map[string]interface{}{
+					"pattern": `^\s`,
+					"match":   "after",
+				},
+			}),
+		},
+		{
+			Id: "logfile-system.auth-default-system",
+			DataStream: &proto.DataStream{
+				Dataset: "system.auth",
+				Type:    "logs",
+			},
+			Source: tests.RequireNewStruct(map[string]interface{}{
```

Comment:
Do these test the complete possible syntax for Filebeat input? Is there a specific integration we could use a reference to try to cover more of the possible inputs? Is there a way to test handling of unexpected configuration entries?

Reply:
Ah, didn't think of trying out bad inputs. Can add a test for that.

I'd like to add more configs outside of the system module, mostly did that because system logs were what I had access to for testing. I'm honestly not sure how comprehensive we want these tests to be, since they're fairly heavy, and basically require building and running the entire beat.

Reply:
> Is there a way to test handling of unexpected configuration entries?

So, now that I actually think about this, I'm not sure how we'd want to test or handle this? Technically, event beats don't handle this particularly well, and a beat will respond to unexpected/invalid input with anything from "ignore, move on" to "refuse to start." Is there any behavior in particular you're interested in testing?

Reply:
With the input being more structured now maybe this is less of a concern.

How are errors reported by the beat when a required configuration is missing? Or in general if the configuration is malformed like the data stream information is missing? Can we report detailed errors in this case?

Reply:
> How are errors reported by the beat when a required configuration is missing? 

Under agent, you mean? Poorly. "the beat is healthy but something isn't working" is a common complaint, presumably because the central management status updater doesn't propagate itself very well, so a lot of errors just don't make their way to any kind of agent endpoint. This is something that needs to be fixed at a beat level.